code/modules/mob/emote.dm

```diff
@@ -112,7 +112,7 @@
 				O.show_message(message, m_type)
 
 /mob/proc/emote_dead(message)
-	if(client.prefs.muted & MUTE_DEADCHAT)
+	if(check_mute(client.ckey, MUTE_DEADCHAT))
```

Comment:
Oh and for whoever asks "Why check the client ckey instead of the mob ckey"

1. Client ckey make more sense to check since its the client controlling said mob
2. Mob ckeys can sometimes be manually edited in and have scuffed casing
3. This works